install/kubernetes/cilium/templates/clustermesh-coredns-mcsapi/job-role.yaml

```diff
@@ -0,0 +1,24 @@
+{{- if and (or .Values.clustermesh.mcsapi.enabled .Values.clustermesh.enableMCSAPISupport) .Values.clustermesh.mcsapi.corednsAutoConfigure.enabled }}
+kind: Role
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: coredns-mcsapi-autoconfig
+  namespace: {{ .Values.clustermesh.mcsapi.corednsAutoConfigure.coredns.namespace }}
+  {{- with .Values.commonLabels }}
+  labels:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+  {{- with .Values.clustermesh.annotations }}
+  annotations:
```

Comment:
These accessory artifacts should also have the `helm.sh/hook` annotation, so that they get automatically removed alongside the job.